nullaway/src/test/java/com/uber/nullaway/NullAwayCoreTests.java

```diff
@@ -623,6 +623,29 @@ public void nullableOnJavaLangVoid() {
         .doTest();
   }
 
+  @Test
+  public void nullableOnJavaLangVoidWithCast() {
+    defaultCompilationHelper
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import javax.annotation.Nullable;",
+            "class Test {",
+            "  Void foo1() {",
+            "    // temporarily, we treat a Void return type as if it was @Nullable Void",
+            "    return (Void)null;",
+            "  }",
+            "  void consumeVoid(Void v) {",
+            "  }",
+            "  @Nullable Void foo2() {",
+            "    // temporarily, we treat a Void argument type as if it was @Nullable Void",
+            "    consumeVoid((Void)null);",
```

Comment:
From the test, it seems that this change makes us treat `(Void) null` as `@NonNull Void` when it is a parameter, not `@Nullable Void` (since here `consumeVoid`'s argument is `@NonNull`).  Am I missing something?

Reply:
What this is meant to change is the interpretation of a `Void` **formal** to automatically mean `@Nullable Void`. See the change to `nonNullPositions` [here](https://github.com/uber/NullAway/pull/613/files#diff-a53186e00f6c045ebdbe4813d3e1ff0a469095e5034d5273571c09334a6cb4e4R1450).

It is not possible at this time to always annotate the formal as `@Nullable`, because, usually, in the cases where we want to pass `Void` around, the formal is of some generic type `T` and what we really would want is to be able to replace `<Void>` with `<@Nullable Void>`, see the example in https://github.com/uber/NullAway/pull/608.

Reply:
Got it thanks! I was thinking about the actual rather than the formal

Reply:
Just doing `consumeVoid(null)` should also have no error right? Can we test that?